### PR TITLE
Fix: Large builds

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

Allows big sizes for builds. Without this limit eas 130MB set by Swoole. After which the container crashed.

This ensures build is done with chunk read/writes.

TODO:

- [x] Ensure git builds respect _APP_FUNCTIONS_SIZE_LIMIT

## Test Plan

- [x] Manual QA

## Related PRs and Issues

* https://github.com/appwrite/appwrite/issues/6245

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

**Comment by Meldiron** [comment]
QA:

Large build:
CleanShot 2023-09-18 at 13 56 57@2x

**Review by lohanidamodar — APPROVED** [review]

**Comment by vermakhushboo** [comment]
If executor limit is 130 MB, why is functions limit only 28 MB in this code?

File: app/workers/builds.php [hunk 1]
```diff
@@ -251,17 +251,24 @@ protected function buildDeployment(GitHub $github, Document $project, Document $
...
+
+                if (substr($tmpDirectory, -1) !== '/') {
+                    $tmpDirectory .= '/';
+                }
+
+                $directorySize = $localDevice->getDirectorySize($tmpDirectory);
+                $functionsSizeLimit = (int) App::getEnv('_APP_FUNCTIONS_SIZE_LIMIT', '30000000');
+                if ($directorySize > $functionsSizeLimit) {
+                    throw new Exception('Repository directory size should be less than ' . number_format($functionsSizeLimit / 1048576, 2) . ' MBs.');
+                }
```
Comment (stnguyen90) [inline app/workers/builds.php:264]: I'm a little concerned about the inconsistency between VCS and manual deployment. In manual deployment, the size limit is on the compressed archive:

https://github.com/appwrite/appwrite/blob/ddb73eb4f60aa807fd0bfe8622da32a69738077a/app/controllers/api/functions.php#L1082-L1120

but here, we're looking at the uncompressed directory size.

Comment (Meldiron) [reply app/workers/builds.php:264]: With manual, compression is done by client. With VCS, compression is done by server.
I am worried we would be spending resources, allowing DOS attack if we moved it lower.

I understand inconsistency concerns, but I think only very very few devs should be hitting the limit on Cloud. And if hit on self-hosted, its so easy to increase that it doesn't matter too much if it's before or after compression size

Comment (stnguyen90) [reply app/workers/builds.php:264]: > I think only very very few devs should be hitting the limit on Cloud. And if hit on self-hosted

@Meldiron, I don't think it's comparison between cloud vs self-hosted. It's manual deployment vs VCS. Both can be done on cloud and self-hosted.

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**